src/PasswordHistory.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+ /**
+  * Helper class that contains method to handle password history validation or
+  * change
+  *
+  * Implemented temporarily as a singleton, should become some kind of service
+  * once GLPI support depencency injection.
+  */
+final class PasswordHistory
+{
+    /**
+     * Max allowed history size
+     */
+    public const MAX_HISTORY_SIZE = 4;
```

Comment:
```suggestion
    public const MAX_HISTORY_SIZE = 5;
```

Reply:
As discussed, this value should no be changed as it only represent the number of password from glpi_users.password_history that must be validated.
More comments details added in 451e6db180.